src/v2/view-builder/views/IdentifierView.js

```diff
@@ -20,28 +20,42 @@ const Body = BaseForm.extend({
 
   saveForm() {
     // Ideally this can be added to a "preSaveForm" handler - but keeping this here for now.
+    this.model.trigger('request');
+
+    this.getDeviceFingerprintPromise()
+      .then(() => this.getProcessCredsPromise())
+      .then(() => {
+        BaseForm.prototype.saveForm.apply(this, arguments);
+      });
+  },
+
+  getDeviceFingerprintPromise() {
     if (!this.settings.get('features.deviceFingerprinting')) {
-      BaseForm.prototype.saveForm.apply(this, arguments);
-      return;
+      return Promise.resolve();
     }
 
-    // Before the XHR is made for "identify", we'll generate this one-time use fingerprint via
-    // a hidden-iframe (similar to authn/v1 flows)
     const fingerprintData = {
       oktaDomainUrl: this.settings.get('baseUrl'),
       element: this.$el,
     };
 
-    this.model.trigger('request');
-
-    DeviceFingerprinting.generateDeviceFingerprint(fingerprintData)
+    return DeviceFingerprinting.generateDeviceFingerprint(fingerprintData)
       .then(fingerprint => {
         this.options.appState.set('deviceFingerprint', fingerprint);
       })
-      .catch(() => { /* Keep going even if device fingerprint fails */ })
-      .finally(() => {
-        BaseForm.prototype.saveForm.apply(this, arguments);
-      });
+      .catch(() => { /* Keep going even if device fingerprint fails */ });
+  },
+
+  getProcessCredsPromise() {
+    const password = this.model.get('credentials.passcode');
+    if (!password) {
+      return Promise.resolve();
+    }
+    const creds = {
+      username: this.model.get('identifier'),
+      password
+    };
+    return this.settings.processCreds(creds);
```

Comment:
I think the checks for identifier/password should either
- go inside `this.settings.processCreds`
or
- not happen at all and just let the user handle it

Reply:
what about avoid to invoke `this.settings.processCreds` if it's identify first flow (only has user name field)

Reply:
That's how I did it originally in this commit: https://github.com/okta/okta-signin-widget/pull/1851/commits/8fa47ee77a4940a66d1c56ab295364a94bfec53d

After I saw I changed it to call processCreds with partial credentials and do the check inside `this.settings.processCreds`. Which approach do we want to take here? I don't feel strongly either way

Reply:
Don't have strong opinion either way but leaning toward to not make changes to `this.settings.processCreds` method. hence the ask.

Reply:
Reverted this change